### ISSUE TITLE
v3.5.0 release plan

### ISSUE THREAD

**Issue by yzhang-gh** [yzhang-gh/vscode-markdown#966]
- [ ] TOC issue when a heading contains math
- [ ] v3.5.0 welcome message
- [ ] revisit possible behavior changes (including test suites) introduced by #885, #890
- [ ] pending #962 decision

**Comment by Lemmingh** [comment]
FYI

[CommonMark 0.30](https://github.com/commonmark/commonmark-spec/releases/tag/0.30) got released a few minutes ago, with mainly clarifications, and subtle changes to boundary conditions.

I think it will not affect our plan for now.

**Comment by yzhang-gh** [comment]
That will be super helpful. Don't hurry and please take your time.

**Comment by yzhang-gh** [comment]
@Lemmingh I have created a `patch` branch and released v3.4.2. (It has been more than a year since the last update.)

Apparently now there is no easy way to merge the two branches (`patch` and `master`). I have tested locally that rebasing `master` on the `patch` is okay (the history is kept linear and the duplicated commits are auto eliminated), while we only need to manually deal with conflicts on `package(-lock).json`. The drawback is it will affect your local branches (need a merge commit).

Or you may have some better ideas?

**Comment by Lemmingh** [comment]
> ideas?

Tag and keep it forever.

It's a common pattern in large projects to create a branch for each `x.y` release. As part of maintenance, critical changes in the main branch will be backported as `x.y.z` patches.

I guess we didn't have such practice because we deprecate old releases as soon as a new version comes out. It sounds not a conflict to me to make release-branches occasionally and have a forward-only life cycle policy meanwhile.

**Comment by yzhang-gh** [comment]
Makes sense. Then I am going to simply cherry-pick the "version number and changelog commit" into the `master` branch.

**Comment by Lemmingh** [comment]
Not sure if I understand you. Do you refer to `CHANGELOG.md`?

**Comment by yzhang-gh** [comment]
Yes (and also the `version` in `package.json`)

**Comment by Lemmingh** [comment]
I guess we can bump the `version` of `master` to `3.5.0`, and copy the `CHANGELOG.md`. Just need to clarify its source in the commit message.

It sounds strange to have a version number that actually doesn't reflect the content.

---

I thought about:

```
📝 Update CHANGELOG

From f791314074e92503249b1688c4fa97e28d8272ac
```

```
🔧 Bump version to 3.5.0
```
